cmd/kubeapps-apis/server/plugins.go

```diff
@@ -298,37 +298,38 @@ func createClientGetter(serveOpts ServeOptions) (func(context.Context) (dynamic.
 
 // createClientGetter takes the required params and returns the closure fuction.
 // it's splitted for testing this fn separately
-func createClientGetterWithParams(inClusterConfig *rest.Config, serveOpts ServeOptions, clustersConfig kube.ClustersConfig) (func(context.Context) (dynamic.Interface, error), error) {
+func createClientGetterWithParams(inClusterConfig *rest.Config, serveOpts ServeOptions, clustersConfig kube.ClustersConfig) (KubernetesClientGetter, error) {
 
 	// return the closure fuction that takes the context, but preserving the required scope,
 	// 'inClusterConfig' and 'config'
-	return func(ctx context.Context) (dynamic.Interface, error) {
+	return func(ctx context.Context) (kubernetes.Interface, dynamic.Interface, error) {
 		var err error
 		token, err := extractToken(ctx)
 		if err != nil {
-			return nil, status.Errorf(codes.Unauthenticated, "invalid authorization metadata: %v", err)
+			return nil, nil, status.Errorf(codes.Unauthenticated, "invalid authorization metadata: %v", err)
 		}
 
-		var client dynamic.Interface
+		var config *rest.Config
 		if !serveOpts.UnsafeUseDemoSA {
 			// We are using the KubeappsClusterName, but if the endpoint was cluster-scoped,
 			// we should pass the cluster name instead
-			restConfig, err := kube.NewClusterConfig(inClusterConfig, token, clustersConfig.KubeappsClusterName, clustersConfig)
+			config, err = kube.NewClusterConfig(inClusterConfig, token, clustersConfig.KubeappsClusterName, clustersConfig)
 			if err != nil {
-				return nil, fmt.Errorf("unable to get clusterConfig: %w", err)
-			}
-			client, err = dynamic.NewForConfig(restConfig)
-			if err != nil {
-				return nil, fmt.Errorf("unable to create dynamic client: %w", err)
+				return nil, nil, fmt.Errorf("unable to get clusterConfig: %w", err)
 			}
 		} else {
 			// Just using the created SA, no user account is used
-			client, err = dynamic.NewForConfig(inClusterConfig)
-			if err != nil {
-				return nil, fmt.Errorf("unable to create dynamic client: %w", err)
-			}
+			config = inClusterConfig
+		}
+		dynamicClient, err := dynamic.NewForConfig(config)
+		if err != nil {
+			return nil, nil, fmt.Errorf("unable to create dynamic client: %w", err)
+		}
+		typedClient, err := kubernetes.NewForConfig(config)
+		if err != nil {
+			return nil, nil, fmt.Errorf("unable to create typed client: %w", err)
```

Comment:
Main change here (other than adding the new client) was just to DRY up the code for creating a client (or clients).

Reply:
Great, thanks for adding it. I initially thought of adding a clientGetter for each type, sth like "clientGetter" and "dynamicClientGetter", but returning both clients is better (assuming there are no more clients out there)